cpp/include/cudf/io_types.h

```diff
@@ -23,27 +23,47 @@
  */
 #pragma once
 
+typedef enum 
```

Comment:
This struct needs a block comment explaining what it is.

Reply:
Done

Reply:
Not following the doxygen template, but since that was helpfully deleted from the repo, I'll let it slide.